component/Header.tsx

```diff
@@ -1,12 +1,16 @@
 import Image from 'next/image'
 import React from 'react'
 import getConfig from 'next/config'
-import { useRouter } from 'next/router'
+import { BetterSocialEventTracking } from 'analytics/analyticsEventTracking'
+import { sendAnalytics } from '@services/analytics/analyticsServices'
 
 const { publicRuntimeConfig } = getConfig()
 
 const Header = ({fixedPosition = true}) => {
-    const toDownloadLink = () => window.open(publicRuntimeConfig.DOWNLOAD_BETTERSOCIAL_APP_URL, '_blank')
+    const toDownloadLink = () => {
+        sendAnalytics(BetterSocialEventTracking.PROFILE_SCREEN_DOWNLOAD_APP_BUTTON_CLICKED)
+        window.open(publicRuntimeConfig.DOWNLOAD_BETTERSOCIAL_APP_URL, '_blank')
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `toDownloadLink` function now includes a call to `sendAnalytics` before opening the download link. This is a good way to track user interactions, but it's important to consider error handling here. If `sendAnalytics` fails for any reason (e.g., network issues), it might prevent the download link from opening. Consider wrapping `sendAnalytics` in a try-catch block or ensure it doesn't throw exceptions that could interrupt the flow.

```diff
    const toDownloadLink = () => {
        try {
            sendAnalytics(BetterSocialEventTracking.PROFILE_SCREEN_DOWNLOAD_APP_BUTTON_CLICKED)
        } catch (error) {
            console.error('Failed to send analytics:', error);
        }
        window.open(publicRuntimeConfig.DOWNLOAD_BETTERSOCIAL_APP_URL, '_blank')
    }
```
This way, even if the analytics fail to send, the user can still proceed with their intended action of downloading the app.


<!-- This is an auto-generated comment by OSS CodeRabbit -->